### ISSUE TITLE
Type only used in interface is not removed 

### ISSUE THREAD

**Issue by duarten** [swc-project/swc#3930]
### Describe the bug

A type-only import is not removed when transpiling TS to JS.

### Input code

```typescript
import { Foo } from 'foo'

interface B extends Foo {
  
}
```

### Playground link

https://play.swc.rs/?version=1.2.151&code=H4sIAAAAAAAAA8vMLcgvKlGoVnDLz1eoVUgrys9VUE%2FLz1fn4srMK0ktSktMTlVwUkitKEnNSykGq6rmUlDgqgUApZukATkAAAA%3D&config=H4sIAAAAAAAAAz2PSwrDMAxE76J1Fm0WXeQEvYZw5eDgH5IMMSF3r5yS7obRzBt0wCYOlgMqshAPJT0r7rCA9kriOFSFCVTMUm50TkC7EmeMb4qVWH6%2BRRiz%2BMJpUJjQ6SVa1pDIeNi0JNTg4DRILEXM9RhldJFXsjyQzI%2F5aYMp5OD7ILiSKpPIP5wwr%2FHuDtadve%2Fl08b9uF64oC%2Fb%2FAIU2c5e7AAAAA%3D%3D

### Expected behavior

The transpiled code should be an empty module.

### Actual behavior

The import is preserved, although it has no runtime meaning in JS.

### Version

1.2.151

**Comment by LeonardSSH** [comment]
I don't think it's really a bug, I mean you can fix it using the `type` keyword.

```ts
import { type Foo } from 'foo';
// or
import type { Foo } from 'foo';

interface B extends Foo { }
```

> [playground link](https://play.swc.rs/?version=1.2.151&code=H4sIAAAAAAAAA8vMLcgvKlGoViipLEhVcMvPV6hVSCvKz1VQT8vPV7fm4srMK0ktSktMTlVwUkitKEnNSykGK6vmUlDgqgUA398XzT8AAAA%3D&config=H4sIAAAAAAAAAz2PSwrDMAxE76J1Fm0WXeQEvYZw5eDgH5IMMSF3r5yS7obRzBt0wCYOlgMqshAPJT0r7rCA9kriOFSFCVTMUm50TkC7EmeMb4qVWH6%2BRRiz%2BMJpUJjQ6SVa1pDIeNi0JNTg4DRILEXM9RhldJFXsjyQzI%2F5aYMp5OD7ILiSKpPIP5wwr%2FHuDtadve%2Fl08b9uF64oC%2Fb%2FAIU2c5e7AAAAA%3D%3D)

**Comment by duarten** [comment]
It's a bug with a workaround, but it is definitely a bug. `tsc` will correctly handle this transformation.

**Comment by magic-akari** [comment]
the expect result is 

```javascript
import {} from 'foo'
```

or 

```javascript
import 'foo';
```

**Comment by duarten** [comment]
The expected result is an empty module, so `export {}`. You can validate that against `tsc`.

**Comment by Austaras** [comment]
It should already be fixed in #3893

**Comment by kdy1** [comment]
Thanks!
